src/battle_controller_player.c

```diff
@@ -1764,11 +1764,38 @@ static void MoveSelectionDisplayMoveType(u32 battler)
 static void MoveSelectionDisplayMoveDescription(u32 battler)
 {
     struct ChooseMoveStruct *moveInfo = (struct ChooseMoveStruct*)(&gBattleResources->bufferA[battler][4]);
-    u16 move = moveInfo->moves[gMoveSelectionCursor[battler]];
-    u16 pwr = gMovesInfo[move].power;
-    u16 acc = gMovesInfo[move].accuracy;
+    u32 move = moveInfo->moves[gMoveSelectionCursor[battler]];
+    u16 pwr = 0;
+    u16 acc = 0;
+    u32 battlerAtk = battler;
+    u32 battlerDef = BATTLE_OPPOSITE(battlerAtk);
+    u32 moveType = gMovesInfo[move].type;
+    u32 atkAbility = GetBattlerAbility(battlerAtk);
+    u32 holdEffectAtk = GetBattlerHoldEffect(battlerAtk, TRUE);
     u8 cat = gMovesInfo[move].category;
 
+    if (B_DYNAMIC_MOVE_DESCRIPTIONS)
+    {
+        u32 moveEffect = gMovesInfo[move].effect;
+
+        if (move == MOVE_NONE || move == 0xFFFF || moveEffect == EFFECT_KNOCK_OFF || moveEffect == EFFECT_BRINE)
+            return;
```

Comment:
this part should probably be included the previous if statement so it branches out to the else statement